File: 1401-1500/1431-kids-with-candies.java
class Solution {
  public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
      int max = 0;
      for(int i=0;i<candies.length;i++) {
          if(candies[i] > max) max = candies[i];
      }
      
      List<Boolean> result = new LinkedList<Boolean>();
      for(int i=0;i<candies.length;i++) {
          Boolean isGreatest = candies[i] + extraCandies >= max;
          result.add(isGreatest);
      }
      
      return result;
  }
}

File: 1401-1500/1431_kids_with_candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := 0
	res := make([]bool, len(candies))

	for _, candies := range candies {
		if max < candies {
			max = candies
		}
	}

	for i, candies := range candies {
		if candies+extraCandies >= max {
			res[i] = true
		}
	}

	return res
}